### PR TITLE
feat: add stream writer

### PR DESCRIPTION
**Pull request by shaoting-huang**
issue: #127

### CODE REVIEW

File: cpp/src/writer/column_group_writer.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+Status ColumnGroupWriter::Write(const std::shared_ptr<arrow::RecordBatch>& record) {
+  if (finished_) {
+    return Status::WriterError("Writer has been closed");
+  }
+  column_group_.AddRecordBatch(record);
+  return Status::OK();
+}
+
+Status ColumnGroupWriter::Flush() {
+  auto status = writer_.WriteTable(*column_group_.Table());
```
Comment (tedxu) [inline cpp/src/writer/column_group_writer.cpp:54]: A chunking or row grouping is expected here, or else, there will be large row groups that won't fit in reader's memory.

File: cpp/include/milvus-storage/writer/column_group.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  size_t size() const;
+
+  GroupId group_id() const;
+
+  Status AddRecordBatch(const std::shared_ptr<arrow::RecordBatch>& batch);
+
+  std::shared_ptr<arrow::Table> Table() const;
+
+  std::shared_ptr<arrow::RecordBatch> GetRecordBatch(size_t index) const;
```
[the remark this answers is not in the document]
Comment (shaoting-huang) [reply cpp/include/milvus-storage/writer/column_group.h:44]: I used size() method. I changed to GetRecordBatchNum

**Pull request merged**